controller/api/locking.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from models import get_etcd_client
+
+log = logging.getLogger(__name__)
+LOCK_PATH_TEMPLATE = '/deis/controller/locks/{name}'
+
+
+class AlreadyLocked(Exception):
+    pass
+
+
+def lock(name, ttl=3600):
+    key = LOCK_PATH_TEMPLATE.format(name=name)
+    value = 'locked'
+    client = get_etcd_client()
```

Comment:
presumably there will be many controller api servers running at once, so has to be a distributed lock